packages/types/src/Metadata/v4/toV5.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017-2019 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Option, Vector } from '../../codec';
+import Text from '../../primitive/Text';
+import MetadataV4 from './Metadata';
+import StorageHasher from '../../primitive/StorageHasher';
+import MetadataV5 from '../v5';
+import { StorageFunctionMetadata as StorageFunctionMetadataV4 } from '../v4/Storage';
+import { DoubleMapType, MapType, StorageFunctionMetadata, StorageFunctionType } from '../v5/Storage';
+
+/**
+ * Convert V4 StorageFunction to V5 StorageFunction
+ */
+function toV5StorageFunction (storageFn: StorageFunctionMetadataV4): StorageFunctionMetadata {
+  const { documentation, fallback, modifier, name, type } = storageFn;
+
+  function toStorageHasher (text: Text): StorageHasher {
+    switch (text.toString()) {
+      case 'blake2_128':
+        return new StorageHasher('Blake2_128');
+      case 'blake2_256':
+        return new StorageHasher('Blake2_256');
+      case 'twox_128':
+        return new StorageHasher('Twox128');
+      case 'twox_256':
+        return new StorageHasher('Twox256');
+      case 'twox_64_concat':
+        return new StorageHasher('Twox64Concat');
+      default:
+        throw new Error(`Invalid Storage hasher: ${text.toString()}`);
+    }
+  }
+
+  const [newType, index] = type.isPlainType
+    ? [type, 0]
+    : type.isMap
+      ? [new MapType({
```

Comment:
I think we can replace this by `[type, 1]` (if TS doesn't complain)

Reply:
`type.asMap` did the trick